### PR TITLE
util typescript

### PR DESCRIPTION
**Pull request by danethurber**
all the typescript!

### CODE REVIEW

File: packages/util/src/create-universal-portal.ts [hunk 1]
```diff
@@ -8,7 +8,7 @@ export function createUniversalPortal(
   container: Element,
   key?: null | string
 ): ReactPortal {
-  if (!canUseDOM) return null
+  if (!canUseDOM()) return null
```
Comment (jaketrent) [inline packages/util/src/create-universal-portal.ts:11]: :)   nice catch.  Yeah, I've seen this at least one other time where exenv's API was a bool and this is a () => bool

File: .eslintignore [hunk 2]
```diff
@@ -3,6 +3,8 @@ node_modules/*
 **/.*
 **/dist/*
 *.dist.js
+*.dist.ts
```
Comment (jaketrent) [inline .eslintignore:6]: I can tell who lints. 

Some time, I'd love for you to review my lint setup on my vim stack.  I think I could utilize that tooling better.

File: packages/util/src/__specs__/combine-fns.spec.ts [hunk 3]
```diff
@@ -6,7 +6,9 @@ test('combineFns()', () => {
   const second = undefined
   const third = null
   const fourth = jest.fn()
+
```
Comment (jaketrent) [inline packages/util/src/__specs__/combine-fns.spec.ts:9]: ooooh yes, I see.....

File: packages/util/src/combine-fns.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
```
Comment (jaketrent) [inline packages/util/src/combine-fns.ts:1]: So are you moving this into util from somewhere? 

This is *in addition* to TS conversion, right?

**Review by jaketrent — COMMENTED** [review]
after root `yarn && yarn build`,

tsc giving me:

```
design-system/packages/util on  refactor/util-ts [$!] ❯ tsc
src/__specs__/storyshots.spec.tsx:11:31 - error TS2345: Argument of type '{ createNodeMock: () => HTMLDivElement; }' is not assignable to parameter of type 'Function | { renderer?: any; serializer?: any; }'.
  Object literal may only specify known properties, and 'createNodeMock' does not exist in type 'Function | { renderer?: any; serializer?: any; }'.

11   test: snapshotWithOptions({ createNodeMock })
                                 ~~~~~~~~~~~~~~

src/__specs__/use-combined-refs.spec.tsx:13:46 - error TS2322: Type '{ value: string; }' is not assignable to type 'undefined'.

13       useImperativeHandle(combinedRef, () => ({
                                                ~~
14         value: 'ref'
   ~~~~~~~~~~~~~~~~~~~~
15       }))
   ~~~~~~~~

  ../../node_modules/@types/react/index.d.ts:1093:79
    1093     function useImperativeHandle<T, R extends T>(ref: Ref<T>|undefined, init: () => R, deps?: DependencyList): void;
                                                                                       ~~~~~~~
    The expected type comes from the return type of this signature.

src/__specs__/use-combined-refs.spec.tsx:26:46 - error TS2322: Type '{ value: string; }' is not assignable to type 'undefined'.

26       useImperativeHandle(combinedRef, () => ({
                                                ~~
27         value: 'ref'
   ~~~~~~~~~~~~~~~~~~~~
28       }))
   ~~~~~~~~

  ../../node_modules/@types/react/index.d.ts:1093:79
    1093     function useImperativeHandle<T, R extends T>(ref: Ref<T>|undefined, init: () => R, deps?: DependencyList): void;
                                                                                       ~~~~~~~
    The expected type comes from the return type of this signature.

src/__specs__/use-combined-refs.spec.tsx:39:46 - error TS2322: Type '{ value: string; }' is not assignable to type 'undefined'.

39       useImperativeHandle(combinedRef, () => ({
                                                ~~
40         value: 'ref'
   ~~~~~~~~~~~~~~~~~~~~
41       }))
   ~~~~~~~~

  ../../node_modules/@types/react/index.d.ts:1093:79
    1093     function useImperativeHandle<T, R extends T>(ref: Ref<T>|undefined, init: () => R, deps?: DependencyList): void;
                                                                                       ~~~~~~~
    The expected type comes from the return type of this signature.


Found 4 errors.
```

File: packages/util/src/combine-fns.ts (hunk 4 above)
Comment (danethurber) [reply packages/util/src/combine-fns.ts:1]: no this was in util before. my git mv, commit, change, commit process doesn't work on diff for some reason

**Comment by jaketrent** [comment]
Dane, not sure if you're planning to do more on this.  Are you getting the same 4 tsc errors that I was?  Do you plan to address them in new commits?

**Comment by danethurber** [comment]
I'll look and see about the errors. I wasn't getting them when I built but I can def try again

File: packages/actionmenu/src/react/index.tsx [hunk 5]
```diff
@@ -52,8 +53,9 @@ interface ActionMenuComponent
 
 export const ActionMenu = React.forwardRef<HTMLUListElement, Props>(
   ({ onClick, onClose, children, origin, ...props }, forwardedRef) => {
-    const fallbackRef = React.useRef()
-    const ref = forwardedRef || fallbackRef
+    const ref = React.useRef<HTMLUListElement>()
+    useImperativeHandle(forwardedRef, () => ref.current)
```
Comment (jaketrent) [inline packages/actionmenu/src/react/index.tsx:57]: Not null problem for you like mine, eh? 🤷

**Review by jaketrent — CHANGES_REQUESTED** [review]
I still get these errors:


```
design-system/packages/util on  refactor/util-ts [⇣$!] ❯ tsc --version
Version 4.0.2
design-system/packages/util on  refactor/util-ts [⇣$!] ❯ tsc
src/__specs__/storyshots.spec.tsx:11:31 - error TS2345: Argument of type '{ createNodeMock: () => HTMLDivElement; }' is not assignable to parameter of type 'Function | { renderer?: any; serializer?: any; }'.
  Object literal may only specify known properties, and 'createNodeMock' does not exist in type 'Function | { renderer?: any; serializer?: any; }'.

11   test: snapshotWithOptions({ createNodeMock })
                                 ~~~~~~~~~~~~~~

src/__specs__/use-combined-refs.spec.tsx:13:46 - error TS2322: Type '{ value: string; }' is not assignable to type 'undefined'.

13       useImperativeHandle(combinedRef, () => ({
                                                ~~
14         value: 'ref'
   ~~~~~~~~~~~~~~~~~~~~
15       }))
   ~~~~~~~~

  ../../node_modules/@types/react/index.d.ts:1093:79
    1093     function useImperativeHandle<T, R extends T>(ref: Ref<T>|undefined, init: () => R, deps?: DependencyList): void;
                                                                                       ~~~~~~~
    The expected type comes from the return type of this signature.

src/__specs__/use-combined-refs.spec.tsx:26:46 - error TS2322: Type '{ value: string; }' is not assignable to type 'undefined'.

26       useImperativeHandle(combinedRef, () => ({
                                                ~~
27         value: 'ref'
   ~~~~~~~~~~~~~~~~~~~~
28       }))
   ~~~~~~~~

  ../../node_modules/@types/react/index.d.ts:1093:79
    1093     function useImperativeHandle<T, R extends T>(ref: Ref<T>|undefined, init: () => R, deps?: DependencyList): void;
                                                                                       ~~~~~~~
    The expected type comes from the return type of this signature.

src/__specs__/use-combined-refs.spec.tsx:39:46 - error TS2322: Type '{ value: string; }' is not assignable to type 'undefined'.

39       useImperativeHandle(combinedRef, () => ({
                                                ~~
40         value: 'ref'
   ~~~~~~~~~~~~~~~~~~~~
41       }))
   ~~~~~~~~

  ../../node_modules/@types/react/index.d.ts:1093:79
    1093     function useImperativeHandle<T, R extends T>(ref: Ref<T>|undefined, init: () => R, deps?: DependencyList): void;
                                                                                       ~~~~~~~
    The expected type comes from the return type of this signature.


Found 4 errors.
```

**Review by jaketrent — APPROVED** [review]
tsc local is clean for me now!

**Pull request merged**